scout/server/blueprints/variant/controllers.py

```diff
@@ -144,6 +145,19 @@ def variant(
     if add_case:
         variant_case(store, case_obj, variant_obj)
 
+    # Fetch ids for grouped cases and prepare alignment dispay
+    case_groups = {}
+    case_group_label = {}
+    if case_obj.get("group"):
+        for group in case_obj.get("group"):
+            case_groups[group] = list(store.cases(group=group))
```

Comment:
This dictionary contains the complete info of all cases in each group. It's quite a lot of data but only the alignment paths are used successively in the igv_button macro. Wouldn't be better to capure only the tracks?

Reply:
I figured its not that much data, and we still catch a few other things as well like display name, other tracks (wigs, beds etc). The alternative would basically be to make a more lightweight case-group-member object for this purpose. Or a general aln-display-case object that would take the unwrapping data added to the case_objs by the alnviewer utils helpers. Both are doable, and probably on the same complexity level as the current reuse-the-case solution.

How does python actually solve the memory management here - does it really deepcopy them, or pass partial references? Languages where you don't do the malloc calls yourself encourages you to think less about that. 😊

Reply:
> How does python actually solve the memory management here - does it really deepcopy them, or pass partial references?

I don't know, you should ask a programmer 😆 Fine, keep it like this!

Reply:
Ok, I dont feel we would particularly benefit elsewhere right now from either of the other two constructs I came up with, so I'll leave it like this. But you do have a good point in general: we could be more explicit (and sparse) with how much we populate the objects for display when we pull them from db. Might be worth thinking about another layer for the purpose of feeding the templates. Perhaps this will be very evident if we go more towards react / json-delivering views.